pkg/manifests/types.go

```diff
@@ -532,6 +547,19 @@ type UserWorkloadConfiguration struct {
 	// Defines the settings for the Thanos Ruler component in user workload
 	// monitoring.
 	ThanosRuler *ThanosRulerConfig `json:"thanosRuler,omitempty"`
+
+	// Defines the list of namespaces for which Prometheus and Thanos Ruler in
+	// user-defined monitoring don't enforce the `namespace` label value in
+	// `PrometheusRule` objects.
+	//
+	// It allows to define recording and alerting rules that can query across
+	// multiple projects instead of deploying identical `PrometheusRule`
```

Comment:
```suggestion
	// multiple projects (including default projects) instead of deploying identical `PrometheusRule`
```

openshift-xxx etc. I think that's how we call them.

Reply:
As Jan mentioned, maybe we should be more explicit in the docs about the fact that enabling this breaks/bypasses the tenancy model as any user that can read metrics for a namespace in NamespacesWithoutLabelEnforcement and write PrometheusRules in that namespace, can now get access to any metric in any namespace.

(maybe it's "safer" to have RulesWithoutLabelEnforcementAllowed disabled by default.)

Reply:
> any user that can read metrics for a namespace in NamespacesWithoutLabelEnforcement and write PrometheusRules in that namespace, can now get access to any metric in any namespace.

hmm not quite possible IIUC. But I agree that it should be clear from the docs that you don't want regular users to have PrometheusRule write access to the `NamespacesWithoutLabelEnforcement` namespace(s).

Reply:
sure but from my POV, it's by design: when you trust a user/service account to create non-namespaced rules, you're effectively providing them access to all metrics.

Reply:
(I agree with you, just suggested that to be more explicit in the doc)

Reply:
how about the suggested addition of `(including default projects)`? , I'd even go with `can query across ALL projects`